RocketChat/Rocket.Chat
# [NEW] Fallback Error component for Engagement Dashboard widgets

## Proposed changes (including videos or screenshots)


As proposed, was added a fallback component to catch errors at Engagement Dashboard widgets individually.
It used an Error boundary to catch `react-query` errors, due to this scenario was necessary to install and use the library   [react-error-boundary](https://github.com/bvaughn/react-error-boundary)  that implements everything and more compared to our ErrorBoundary component, the main reason was to capture Query errors and the implementation with `react-query` library.

**New layout:**

Before:
![image](https://user-images.githubusercontent.com/20212776/184968003-607eda93-ae3f-406c-a775-becd2720a607.png)

After:
![image](https://user-images.githubusercontent.com/20212776/184970152-25a425f3-6aad-4620-b1c1-5f8c8bb35fbb.png)




## Issue(s)


## Steps to test or reproduce

1 - Login as an Admin in an Enterprise Edition Rocket.chat
2 - Go to administration > Engagement Dashboard
3 - Simulate an error direct in the code or with the browser Network tab
4 - You should be able to see the fallback component and a button to refetch the query

## Further comments

TODO: 

- [x] Add e2e tests to engagement dashboard page/widgets to simulate an error and refetch the query **(The e2e tests were removed to due an issue to test EE features)**
- [x] Remove old ErrorBoundary component and replace with the new library  [react-error-boundary](https://github.com/bvaughn/react-error-boundary)
